### ISSUE TITLE
-np: command not found

### ISSUE THREAD

**Issue by HamzaAsiff** [FedML-AI/FedML#402]
I followed the complete Readme guide to run the run_simulation.sh file but whenever I try to run the "run_simulation.sh" file, I get the following error:

"run_simulation.sh: line 6: -np: command not found"

Terminal: Git Bash
Command:  bash run_simulation.sh 2
OS: Windows

Note: MPI and all other dependencies are installed correctly.

**Comment by zuluzazu** [comment]
@HamzaAsiff Can you please tell in which application did you face this issue? Was it fednlp or fedcv or anything else?

**Comment by HamzaAsiff** [comment]
It was Fedcv.
Mainly I'm trying to train Yolov5 on Coco dataset.

**Comment by beiyuouo** [comment]
Hi @HamzaAsiff. This bash file cannot run in Windows platform, we will fix it soon.

You can run the app by following command:

```bash
mpiexec -n 3 python main_fedml_object_detection.py --cf config/simulation/fedml_config.yaml
``

**Comment by HamzaAsiff** [comment]
After the 1st Training on server side I'm getting the following:

[FedML-Server(0) @device-id-0] [Mon, 25 Jul 2022 09:50:14] [INFO] [client_manager.py:153:send_message] Sending message (type 3) to server
Sending message (type 3) to server
Sending message (type 3) to server
[FedML-Server(0) @device-id-0] [Mon, 25 Jul 2022 09:50:14] [INFO] [server_manager.py:157:receive_message] receive_message. rank_id = 0, msg_type = 3.
receive_message. rank_id = 0, msg_type = 3.
receive_message. rank_id = 0, msg_type = 3.
[FedML-Server(0) @device-id-0] [Mon, 25 Jul 2022 09:50:14] [INFO] [FedAVGAggregator.py:53:add_local_trained_result] add_model. index = 0
add_model. index = 0
add_model. index = 0
[FedML-Server(0) @device-id-0] [Mon, 25 Jul 2022 09:50:14] [INFO] [FedAvgServerManager.py:62:handle_message_receive_model_from_client] b_all_received = True
b_all_received = True
b_all_received = True
[FedML-Server(0) @device-id-0] [Mon, 25 Jul 2022 09:50:14] [INFO] [FedAVGAggregator.py:77:aggregate] len of self.model_dict[idx] = 2
len of self.model_dict[idx] = 2
len of self.model_dict[idx] = 2

[FedML-Server(0) @device-id-0] [Mon, 25 Jul 2022 09:50:14] [ERROR] [mlops_runtime_log.py:32:handle_exception] Uncaught exception
Traceback (most recent call last):
  File "main_fedml_object_detection.py", line 17, in <module>
    fedml_runner = FedMLRunner(args, device, dataset, model, trainer)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\runner.py", line 27, in __init__
    args, device, dataset, model, client_trainer, server_aggregator
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\runner.py", line 38, in _init_simulation_runner
    runner = SimulatorMPI(args, device, dataset, model, client_trainer, server_aggregator)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\simulator.py", line 85, in __init__
    preprocessed_sampling_lists=None,
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAvgAPI.py", line 52, in FedML_FedAvg_distributed
    preprocessed_sampling_lists,
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAvgAPI.py", line 128, in init_server
    server_manager.run()
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAvgServerManager.py", line 30, in run
    super().run()
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\core\distributed\server\server_manager.py", line 140, in run
    self.com_manager.handle_receive_message()
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\core\distributed\communication\mpi\com_manager.py", line 99, in handle_receive_message
    self.notify(msg_params)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\core\distributed\communication\mpi\com_manager.py", line 121, in notify
    observer.receive_message(msg_type, msg_params)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\core\distributed\server\server_manager.py", line 160, in receive_message
    handler_callback_func(msg_params)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAvgServerManager.py", line 64, in handle_message_receive_model_from_client
    global_model_params = self.aggregator.aggregate()
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAVGAggregator.py", line 82, in aggregate
    averaged_params = self._fedavg_aggregation_(model_list)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAVGAggregator.py", line 102, in _fedavg_aggregation_
    averaged_params[k] /= training_num
RuntimeError: result type Float can't be cast to the desired output type Long
Uncaught exception
Traceback (most recent call last):
  File "main_fedml_object_detection.py", line 17, in <module>
    fedml_runner = FedMLRunner(args, device, dataset, model, trainer)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\runner.py", line 27, in __init__
    args, device, dataset, model, client_trainer, server_aggregator
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\runner.py", line 38, in _init_simulation_runner
    runner = SimulatorMPI(args, device, dataset, model, client_trainer, server_aggregator)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\simulator.py", line 85, in __init__
    preprocessed_sampling_lists=None,
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAvgAPI.py", line 52, in FedML_FedAvg_distributed
    preprocessed_sampling_lists,
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAvgAPI.py", line 128, in init_server
    server_manager.run()
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAvgServerManager.py", line 30, in run
    super().run()
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\core\distributed\server\server_manager.py", line 140, in run
    self.com_manager.handle_receive_message()
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\core\distributed\communication\mpi\com_manager.py", line 99, in handle_receive_message
    self.notify(msg_params)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\core\distributed\communication\mpi\com_manager.py", line 121, in notify
    observer.receive_message(msg_type, msg_params)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\core\distributed\server\server_manager.py", line 160, in receive_message
    handler_callback_func(msg_params)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAvgServerManager.py", line 64, in handle_message_receive_model_from_client
    global_model_params = self.aggregator.aggregate()
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAVGAggregator.py", line 82, in aggregate
    averaged_params = self._fedavg_aggregation_(model_list)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAVGAggregator.py", line 102, in _fedavg_aggregation_
    averaged_params[k] /= training_num
RuntimeError: result type Float can't be cast to the desired output type Long
Uncaught exception
Traceback (most recent call last):
  File "main_fedml_object_detection.py", line 17, in <module>
    fedml_runner = FedMLRunner(args, device, dataset, model, trainer)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\runner.py", line 27, in __init__
    args, device, dataset, model, client_trainer, server_aggregator
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\runner.py", line 38, in _init_simulation_runner
    runner = SimulatorMPI(args, device, dataset, model, client_trainer, server_aggregator)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\simulator.py", line 85, in __init__
    preprocessed_sampling_lists=None,
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAvgAPI.py", line 52, in FedML_FedAvg_distributed
    preprocessed_sampling_lists,
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAvgAPI.py", line 128, in init_server
    server_manager.run()
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAvgServerManager.py", line 30, in run
    super().run()
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\core\distributed\server\server_manager.py", line 140, in run
    self.com_manager.handle_receive_message()
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\core\distributed\communication\mpi\com_manager.py", line 99, in handle_receive_message
    self.notify(msg_params)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\core\distributed\communication\mpi\com_manager.py", line 121, in notify
    observer.receive_message(msg_type, msg_params)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\core\distributed\server\server_manager.py", line 160, in receive_message
    handler_callback_func(msg_params)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAvgServerManager.py", line 64, in handle_message_receive_model_from_client
    global_model_params = self.aggregator.aggregate()
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAVGAggregator.py", line 82, in aggregate
    averaged_params = self._fedavg_aggregation_(model_list)
  File "C:\Users\Hamza\anaconda3\envs\FL\lib\site-packages\fedml\simulation\mpi\fedavg\FedAVGAggregator.py", line 102, in _fedavg_aggregation_
    averaged_params[k] /= training_num
RuntimeError: result type Float can't be cast to the desired output type Long

**Comment by beiyuouo** [comment]
@HamzaAsiff Could you run `fedml env` to provide more context information

**Comment by ATEDUST** [comment]
> @HamzaAsiff Could you run `fedml env` to provide more context information

can I ask you how to fix "-np: command not found " ? I can't fix in the version of 0.7339
